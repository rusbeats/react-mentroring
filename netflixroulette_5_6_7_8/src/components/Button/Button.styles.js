import constants from '../../utils/constants';
import styled from 'styled-components';

const { colors, getPaddingBySize } = constants;

export const StyledButton = styled.button.attrs(({ size, active, margin, disabled }) => ({
    padding: getPaddingBySize(size),
    backgroundColor: active ? colors.lightPink : colors.grey,
    margin: margin ? '20px' : '0',
    disabled
}))`
    color: ${colors.white};
    background-color: ${props => props.backgroundColor};
    border: none;
    padding: ${props => props.padding};
    text-align: center;
    cursor: pointer;
    text-transform: uppercase;
    font-size: 16px;
    font-weight: bold;
    margin-right: ${props => props.margin};
    border-radius: 3px;
`;
