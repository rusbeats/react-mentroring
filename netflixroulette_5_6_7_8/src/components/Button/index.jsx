import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton } from './Button.styles';

const Button = ({ size, active, margin, children, disabled, onClick }) => {
    return (
        <StyledButton size={size} active={active} margin={margin} disabled={disabled} onClick={onClick}>
            {children}
        </StyledButton>
    );
};

StyledButton.displayName = 'Button';

Button.propTypes = {
    size: PropTypes.string,
    active: PropTypes.bool,
    margin: PropTypes.bool,
    children: PropTypes.node,
    disabled: PropTypes.bool,
    onClick: PropTypes.func
};

export default Button;
