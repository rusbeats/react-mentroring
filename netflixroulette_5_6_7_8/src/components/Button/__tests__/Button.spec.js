import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';
import { StyledButton } from '../Button.styles';

const PROPS = {
    size: 'small',
    text: 'Button text'
};

describe('Button', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(
            <Button size={PROPS.size} active margin>
                {PROPS.text}
            </Button>
        );
        expect(component).toMatchSnapshot();
    });

    it('should be active and with margin', () => {
        const component = shallow(<StyledButton size={PROPS.size} active margin />);
        expect(component.props()).toEqual(expect.objectContaining({ active: true, margin: true, size: 'small' }));
    });

    it('should be not active and without margin', () => {
        const component = shallow(<StyledButton size={PROPS.size} active={false} margin={false} />);
        expect(component.props()).toEqual(expect.objectContaining({ active: false, margin: false, size: 'small' }));
    });
});
