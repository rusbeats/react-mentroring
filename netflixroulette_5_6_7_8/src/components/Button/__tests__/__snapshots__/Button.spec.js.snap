// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly (snapshot) 1`] = `
<Button
  active={true}
  margin={true}
  size="small"
>
  Button text
</Button>
`;
